test/integration/connection-monitoring-and-pooling/connection.test.ts

```diff
@@ -35,15 +33,15 @@ describe('Connection', function () {
     it('should execute a command against a server', {
       metadata: { requires: { apiVersion: false, topology: '!load-balanced' } },
       test: async function () {
-        const connectOptions: Partial<ConnectionOptions> = {
+        const connectOptions: ConnectionOptions = {
```

Comment:
Why do we remove `Partial` here? To get rid of those castings later `connectOptions as any as ConnectionOptions`? Because connectOptions has indeed partial type here.

Reply:
I rounded out the type so its no longer partial :)

Rather than cast away a problem we should mark why we are able to get away with violating TS but converting tests to TS is sometimes done to large files and we can leave behind type errors that don't actually break the code at runtime.